### PR TITLE
Wrapped tile content on builds and pods pages into 2 column layout

### PR DESCRIPTION
**Pull request by sg00dwin**
for better presentation by using available space

Update UI screenshots for browse builds and pods pages

### CODE REVIEW

**Comment by spadgett** [comment]
@sg00dwin we have an issue to collapse old builds and allow click to expand. We should chat about how this would look with 2 columns.

**Comment by spadgett** [comment]
See #2979

**Comment by spadgett** [comment]
Disregard my comment above. I was thinking you had two columns of tiles, not two columns inside the tile.

File: assets/app/views/builds.html [hunk 1]
```diff
@@ -143,11 +141,15 @@
             </dl>
           </div>
         </div>
-        <div class="well" style="margin-bottom: 10px;" ng-repeat="build in buildsByBuildConfig[buildConfigName] | orderObjectsByDate : true">
+        <div class="well build-log" ng-repeat="build in buildsByBuildConfig[buildConfigName] | orderObjectsByDate : true">
```
Comment (spadgett) [inline assets/app/views/builds.html]: I'd suggest a different class name, maybe just `build`. Eventually we want to surface the actual build log in the UI, which is something else.

File: assets/app/views/builds.html [hunk 2]
```diff
@@ -143,11 +141,15 @@
...
           <h3>{{build.metadata.name}}</h3>
-          <div>Created: <relative-timestamp timestamp="build.metadata.creationTimestamp"></relative-timestamp></div>
-          <div>
-            <span style="margin-right: 5px;">Status: {{build.status.phase}}</span>
+          <div class="row">
+            <div class="col-md-6">
+              <div class="log-meta">
```
Comment (spadgett) [inline assets/app/views/builds.html]: Same comment as above on class names. Maybe `build-details` and `build-details-label`?

File: assets/app/views/builds.html [hunk 3]
```diff
@@ -143,11 +141,15 @@
...
+          <span class="log-meta-title">Created:</span><relative-timestamp timestamp="build.metadata.creationTimestamp"></relative-timestamp>
+              </div>
+          <div class="log-meta">
+            <span class="log-meta-title">Status:</span>
```
Comment (spadgett) [inline assets/app/views/builds.html]: Indentation is a little off here. I'm having trouble seeing the structure and what tags are supposed to be closed where.

File: assets/app/views/builds.html [hunk 4]
```diff
@@ -211,11 +218,13 @@ <h2 ng-if="buildConfigName != ''">
...
-        <div class="well" ng-repeat="build in blds | orderObjectsByDate : true">
+        <div class="well build-log" ng-repeat="build in blds | orderObjectsByDate : true">
           <h3>{{build.metadata.name}}</h3>
-          <div>Created: <relative-timestamp timestamp="build.metadata.creationTimestamp"></relative-timestamp></div>            
-          <div>
-            <span style="margin-right: 5px;">Status: {{build.status.phase}}</span>          
+          <div class="row">
+            <div class="col-md-6">
+              <div class="log-meta"><span class="log-meta-title">Created:</span><relative-timestamp timestamp="build.metadata.creationTimestamp"></relative-timestamp></div>            
+          <div class="log-meta">
```
Comment (spadgett) [inline assets/app/views/builds.html]: Confusing indentation.

File: assets/app/views/pods.html [hunk 5]
```diff
@@ -10,56 +10,70 @@
           <em>{{emptyMessage}}</em>
         </div>
       </div>      
-      <div ng-repeat="pod in pods | orderObjectsByDate : true" class="tile">
+      <div ng-repeat="pod in pods | orderObjectsByDate : true" class="tile pod-log">
```
Comment (spadgett) [inline assets/app/views/pods.html]: Maybe `pod` instead of `pod-log`?

File: assets/app/views/pods.html [hunk 6]
```diff
@@ -10,56 +10,70 @@
...
-      </div>
+            <div class="log-meta">
+              <span class="log-meta-title">Node:</span>{{pod.spec.host || 'unknown'}} <span ng-if="pod.status.hostIP && pod.spec.host != pod.status.hostIP">({{pod.status.hostIP}})</span>
+            </div>
+            <div class="log-meta">
+              <span class="log-meta-title">IP on node:</span>{{pod.status.podIP || 'unknown'}}
+            </div>
+            <div class="log-meta">
+              <span class="log-meta-title">Restart policy:</span>
+              <span ng-if="pod.spec.restartPolicy.always">
```
Comment (spadgett) [inline assets/app/views/pods.html]: Looks like a bad merge. My fix for #3213 was lost.

**Comment by spadgett** [comment]
Looks much better overall. Does this collapse to one column at mobile resolutions?

The `Rebuild` button looks strange to me in the second column, although I didn't like its placement before either.

File: assets/app/views/builds.html [hunk 7]
```diff
@@ -143,62 +141,77 @@
...
+            <div class="col-md-7 col-sm-7 col-xs-7">
+              <h3>{{build.metadata.name}}</h3>
             </div>
-            <div ng-switch-when="Docker">
-              <div ng-if="build.spec.strategy.dockerStrategy.from">
-                Builder image: {{build.spec.strategy.dockerStrategy.from | imageObjectRef : build.metadata.namespace}}
+            <div class="col-md-4 col-sm-4 col-xs-offset-1 col-xs-4 text-right">
+              <div>
+                <button class="btn btn-default" ng-click="cloneBuild(build.metadata.name)" ng-disabled="(buildConfigBuildsInProgress[buildConfigName] | hashSize) > 0">Rebuild</button>
               </div>
```
Comment (sg00dwin) [inline assets/app/views/builds.html]: Moved "Rebuild" button to be aligned with build name - upper right

https://raw.githubusercontent.com/sg00dwin/origin/browse-view-tile-2column/docs/screenshots/browse-builds.png

File: assets/app/views/builds.html (hunk 1 above)
Comment (sg00dwin) [inline assets/app/views/builds.html]: **switched class names**
build-log to build-unit (felt that 'build' was to generic)
pod-log to pod-unit (felt that 'build' was to generic)
log-meta to build-detail and pod-detail
log-meta-title to build-detail-label and pod-detail-label

File: assets/app/views/builds.html (hunk 3 above)
Comment (sg00dwin) [inline assets/app/views/builds.html]: Brought indentation into alignment

File: assets/app/views/pods.html (hunk 6 above)
Comment (sg00dwin) [inline assets/app/views/pods.html]: updated with your fix https://github.com/openshift/origin/pull/3213

File: assets/app/views/pods.html (hunk 5 above)
Comment (sg00dwin) [inline assets/app/views/pods.html]: see rename note above

File: assets/app/styles/_components.less [hunk 8]
```diff
@@ -294,6 +294,17 @@
   }
 }
 
+.build-unit, .pod-unit {
```
Comment (spadgett) [inline assets/app/styles/_components.less]: Still not crazy about the name. `build-well` and `pod-well`?

**Comment by spadgett** [comment]
Spacing is a little tight here:

openshift_web_console

Rebuild button would also look better vertically aligned with the name heading at its left.

**Comment by spadgett** [comment]
I'd like to come back at some point to update the deployments page. Even if we don't use a two-column layout, the label styles now look different. Not necessarily in this pull.

**Comment by sg00dwin** [comment]
@spadgett 
Adjusted spacing of rebuild btn and updated parent class names

**Comment by spadgett** [comment]
LGTM

@liggitt

**Comment by sg00dwin** [comment]
@liggitt any update on this?

**Comment by liggitt** [comment]
@sg00dwin, sorry I didn't get to this last week. @spadgett will take a final look and we'll get it merged

**Comment by spadgett** [comment]
Last nitty comment, maybe for a follow-on. I feel like if we have this much whitespace after the label, the values should be left-aligned.

screen_shot_2015-07-07_at_4_07_00_pm

Otherwise, LGTM

@liggitt

**Comment by liggitt** [comment]
[merge]

**Comment by spadgett** [comment]
```
--- FAIL: TestAuthorizationSubjectAccessReview-2 (8.94s)
	authorization_test.go:295: cluster admin told danny can get project default: from review
			&api.SubjectAccessReview{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Verb:"get", Resource:"projects", User:"danny", Groups:util.StringSet(nil), Content:runtime.EmbeddedObject{Object:runtime.Object(nil)}, ResourceName:""}
		expected
			&api.SubjectAccessReviewResponse{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Namespace:"default", Allowed:true, Reason:"allowed by rule in default"}
		got
			&api.SubjectAccessReviewResponse{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Namespace:"default", Allowed:false, Reason:"User \"danny\" cannot get projects in project \"default\""}
FAIL
```

```
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1505) [receiver=3.0.6]
==> openshiftdev: Unable to download artifacts
Build step 'Execute shell' marked build as failure
```

**Comment by sg00dwin** [comment]
@spadgett I don't understand that failure. Is their something I need to change?

**Comment by spadgett** [comment]
@sg00dwin no, it's unrelated to your change. Put it in a comment because it looked spurious

**Pull request merged**